### PR TITLE
Added implementation of decorators in Python

### LINKED ISSUE #109: [tutorial] Python Decorators
**Implementing Decorators in Python**
...

**Do you want to be assigned to work on this?**
- [x] YES
- [ ] NO

**Explanation with Use Cases**
...

### PR DESCRIPTION
**Pull request by amulyavarshney**
### 2. Description
Tutorial for Implementation of Decorators in Python with a use case.

### 3. Fill in checklist by marking [x]

- [x] I've read the CONTRIBUTING.md
- [x] I was assigned to this issue
- [x] My code is formatted using required linting
- [x] I've run my code locally and checked it works

### CODE REVIEW

**Comment by funbeedev** [comment]
@amulyavarshney We have a lot of PRs to review today so we'll mark as hacktoberfest-accepted for now. Will fully review later and request for changes if needed. Thanks!

**Comment by amulyavarshney** [comment]
@funbeedev Please Review my PR and add it to Hackoctoberfest contribution.

**Comment by amulyavarshney** [comment]
@funbeedev it is still not accepted.

**Comment by funbeedev** [comment]
@amulyavarshney Missed that. Accepted now :)

File: decorators/timer.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+# defining a function to decorate with timer.
+def run():
+    for i in range(10):
+        print(i)
+
+# calling the function which is decorated with timer.
+run()
+
+# ------------------------------------------------------------------------------------
+# Challenge: Create a decorator to wrap a function which logs the function call with the timestamp when it is called.
```
Comment (sabderemane) [inline decorators/timer.py:31]: I'm not sure I understand the challenge, what did you mean ? IMO something easier will be better for a first decorator like print something before and after the function or similar.

**Review by sabderemane — CHANGES_REQUESTED** [review]

File: decorators/timer.py (hunk 1 above)
Comment (amulyavarshney) [reply decorators/timer.py]: Challenge Updated. Thanks!!

**Review by amulyavarshney — COMMENTED** [review]
Challenge has been updated. Please review.

**Review by sabderemane — APPROVED** [review]

**Comment by sabderemane** [comment]
Thank you for your contribution @amulyavarshney ! 🎉

**Pull request merged**